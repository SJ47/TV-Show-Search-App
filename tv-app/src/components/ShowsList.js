import React from 'react'

// This component will render the view of a list of shows matching a search string input by the user
const ShowsList = ({ shows, onSelectedShow }) => {

    // Iterate through all the shows returned and render out the html with a clickable image for each show listed
    const showNames = shows.map((show) => {
        // If show object or show image do not exist, just return null
        if (!show || !show.show.image) {
            return (null)
        }

        // Replace any other null data with a placeholder so as not to error out
        // if (!show.show.image) {
        //     show.show.image = { medium: "https://st4.depositphotos.com/14953852/24787/v/600/depositphotos_247872612-stock-illustration-no-image-available-icon-vector.jpg" }
        // }

        if (!show.show.network) {
            show.show.network = { name: "unknown" }
        }

        if (!show.show.rating.average) {
            show.show.rating.average = "na"
        }

        return (
            <li onClick={() => { onSelectedShow(show) }} key={show.show.id}>
                <img className="card-img" src={show.show.image.medium} width="150" height="200px" alt="">
                </img>
            </li >
        )
    })

    // If no results are found, just display a no results found message
    let mess = "";
    if (showNames.length === 0) {
        mess = "No results found"
    }

    // Render the views
    return (
        <>
            <div className="left-column">
                <h2 className="search-results-header">Search Results</h2>
                <hr />
                <ul>{showNames}{mess}</ul>

            </div>
        </>
    )
}

export default ShowsList
