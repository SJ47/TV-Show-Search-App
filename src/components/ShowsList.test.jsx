import { render } from '@testing-library/react';
import ShowsList from './ShowsList';

const shows = [
  {
    "score": 0.70166075,
    "show": {
    "id": 83,
    "url": "https://www.tvmaze.com/shows/83/the-simpsons",
    "name": "The Simpsons",
    "type": "Animation",
    "language": "English",
    "genres": [
    "Comedy",
    "Family"
    ],
    "status": "Running",
    "runtime": 30,
    "averageRuntime": 30,
    "premiered": "1989-12-17",
    "ended": null,
    "officialSite": "http://www.fox.com/the-simpsons/full-episodes",
    "schedule": {
    "time": "20:00",
    "days": [
    "Sunday"
    ]
    },
    "rating": {
    "average": 8.1
    },
    "weight": 100,
    "network": {
    "id": 4,
    "name": "FOX",
    "country": {
    "name": "United States",
    "code": "US",
    "timezone": "America/New_York"
    },
    "officialSite": "https://www.fox.com/"
    },
    "webChannel": null,
    "dvdCountry": null,
    "externals": {
    "tvrage": 6190,
    "thetvdb": 71663,
    "imdb": "tt0096697"
    },
    "image": {
    "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/423/1058261.jpg",
    "original": "https://static.tvmaze.com/uploads/images/original_untouched/423/1058261.jpg"
    },
    "summary": "<p><b>The Simpsons</b> is the longest running scripted show in US television history. It captures the adventures of Homer, Marge, Maggie, Bart and Lisa who are living in a fictional town called Springfield.</p>",
    "updated": 1678060033,
    "_links": {
    "self": {
    "href": "https://api.tvmaze.com/shows/83"
    },
    "previousepisode": {
    "href": "https://api.tvmaze.com/episodes/2493455"
    },
    "nextepisode": {
    "href": "https://api.tvmaze.com/episodes/2495308"
    }
    }
    }
    },
]
const mockOnSelectedShow = jest.fn();

describe('ShowsList', () => {
  it('should render styles correctly', () => {
    const { container } = render(
        <ShowsList shows ={shows} onSelectedShow={mockOnSelectedShow} />
    );
    expect(container).toMatchSnapshot();
  });
});
