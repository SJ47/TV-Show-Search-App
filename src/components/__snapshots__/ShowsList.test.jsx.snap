// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowsList should render styles correctly 1`] = `
<div>
  <div
    class="left-column"
  >
    <h2
      class="search-results-header"
    >
      Search Results
    </h2>
    <hr />
    <ul>
      <li>
        <img
          alt=""
          class="card-img"
          height="200px"
          src="https://static.tvmaze.com/uploads/images/medium_portrait/423/1058261.jpg"
          width="150"
        />
      </li>
      
    </ul>
  </div>
</div>
`;
